libpod/container_top.go

```diff
@@ -0,0 +1,177 @@
+package libpod
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+
+	"encoding/json"
+	"github.com/pkg/errors"
+	"github.com/projectatomic/libpod/utils"
+	"github.com/sirupsen/logrus"
+)
+
+// This can be removed is stats is merged
+// CGroupParent - should be used from stats
+var CGroupParent = "/libpod_parent"
+
+// GetContainerPids reads sysfs to obtain the pids associated with the container's cgroup
+func (c *Container) GetContainerPids() ([]string, error) {
+	taskFile := filepath.Join("/sys/fs/cgroup/pids", CGroupParent, fmt.Sprintf("libpod-conmon-%s", c.ID()), c.ID(), "tasks")
```

Comment:
Can you add a lock/unlock/sync here, and then return an error if the container isn't running?

Reply:
This might break GetContainerPidInformation which needs an unlocked version, so maybe just make this getContainerPids() and have a GetContainerPids() that locks, syncs, and then calls getContainerPids